src/features/shared-link-settings-modal/SharedLinkSettingsModal.js

```diff
@@ -41,6 +41,8 @@ class SharedLinkSettingsModal extends Component {
         /** Server URL prefix for vanity URL preview; should be something like http://company.box.com/v/ */
         serverURL: PropTypes.string.isRequired,
         vanityNameError: PropTypes.string,
+        /** Add warning about public nature of vanity URLs */
+        warnOnPublic: PropTypes.bool,
```

Comment:
This needs to be a configurable option? This would mean that another client like Notes may not show this warning, even after upgrading to the latest version, right?

Reply:
Oh, never mind... I see now with the other PR that it's based on the shared link access level.